src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/OSFileSystem.cs

```diff
@@ -45,6 +47,55 @@ public void ReadFirstLine(FileInfo file, BufferWriter<char> destination)
     public void ReadAll(FileInfo file, BufferWriter<char> destination)
         => ReadUntilTerminatorOrEnd(file, destination, null);
 
+    public IEnumerable<ReadOnlyMemory<char>> ReadAllByLines(FileInfo file, BufferWriter<char> destination)
```

Comment:
Is there a reason we don't use API provided by the .NET runtime?

Reply:
The implementation is similar to the ReadUntilTerminatorOrEnd method located in the same file, and what we try to achieve is minimize memory allocations here. Both implementations have been reviewed and used for a while in the internal repo.

Are you referring to some specific .NET runtime API?

Reply:
I've been thinking of [`File.ReadAllLines`](https://learn.microsoft.com/dotnet/api/system.io.file.readalllines), but I see it returns `string[]`, which would be allocating.